Radzen.Blazor/RadzenDataGrid.razor

```diff
@@ -829,38 +829,51 @@
     {
         get
         {
-            var view = base.View.Where<TItem>(columns);
-            var orderBy = GetOrderBy();
-
-            if (!string.IsNullOrEmpty(orderBy))
-            {
-                if (typeof(TItem) == typeof(object))
+            bool serverSideDataOperations = LoadData.HasDelegate;
+            var view = base.View;
+            if (!serverSideDataOperations)
+            {
+                var columnsWithNoCustomFiltering = columns.Where(c => c.CustomFiltering == null);
+                view = view.Where<TItem>(columnsWithNoCustomFiltering);
+
+                var columnsWithCustomFiltering = columns.Where(c => c.CustomFiltering != null);
+                if (columnsWithCustomFiltering.Any())
+                {
+                    view = view.AsEnumerable<TItem>().Where<TItem>(x => ApplyDataCustomFiltering(x, columnsWithCustomFiltering)).AsQueryable();

```

Comment:
AsEnumerable<TItem>() will get all data and will brake completely IQueryable support.

Reply:
I assume that the Custom filter can be executed with only in-memory data, probably no ORM exist that can execute the requirements that may be required by the user in the code of CustomFilter.

A custom filter can be placed after OrderBy so that OrderBy is also quered on the database side. But it seems to me that in this case it will not be "real" filtering, since there is a possibility that the user in the custom code will add a limit on the amount of output data as one of the filtering conditions. And if the CustomFilter is located after the OrderBy, then the output dataset will not be equal to the expected one.

Reply:
Hey @viordash,

I'm afraid that we cannot merge such custom filtering. Both in-memory (EnumerableQuery or so called LINQ to Objects) and database filtering can be done using the same approach - LINQ expressions, however this is not the case with this pull request. More info about Expression class can be found here:
https://docs.microsoft.com/en-us/dotnet/api/system.linq.expressions.expression?view=net-5.0

I see what you are after, FilterExpression property for the DataGridColumn that can work both for in-memory and database filtering and we will do our best to provide such in some of our next versions.